vspace/src/main/java/edu/asu/diging/vspace/core/data/SlideRepository.java

```diff
@@ -25,5 +28,16 @@ public interface SlideRepository extends PagingAndSortingRepository<Slide, Strin
 
     Page<ISlide> findDistinctByNameContainingOrDescriptionContaining(Pageable requestedPage, String name,
             String description);
+    
+    @Query(value="SELECT d.Slide_Id FROM Sequence_Slides d, Module m WHERE d.Sequence_Id = m.startSequence_id", nativeQuery = true)
+    Set<String> findAllSlidesFromStartSequences();
+    
+    @Query("SELECT DISTINCT s from Slide s where  s.module.id IN "
+            + "(SELECT DISTINCT module.id from Module module JOIN ModuleLink moduleLink ON module.id = moduleLink.module.id JOIN Space space "
+            + "on space.id = moduleLink.space.id WHERE space.spaceStatus = ?3 AND module.moduleStatus = ?4) "
+            + "AND s.id IN (Select d.id from Sequence ss join ss.slides d , Module m where m.startSequence.id = ss.id)"
+            + "AND s.name like %?1% OR s.description like %?2%")
+    Page<ISlide> findDistinctByNameContainingOrDescriptionContainingLinkedToSpace(Pageable requestedPage, String name,
```

Comment:
since this method is using a query instead of generating the query by the method name, let's give it a more handy name.